cogs/displayinfo.py

```diff
@@ -33,85 +33,113 @@ class DisplayInfo(commands.Cog):
-                self.current_guild_name = mem_before.guild.name
-                self.current_collection = newCollection               
-                newCollection.insert_one({}) # Create a new blank document in the new Collection
-                self.timers = self.current_collection.find_one() # Load blank document from the MongoDB collection
-            else:
-                self.current_collection = db[mem_before.guild.name] # If the collection exists, get the existing collection for the guild
-                self.current_guild_name = mem_before.guild.name
-                self.timers = self.current_collection.find_one() # Load existing document from the MongoDB collection
+
+        if mem_before.status is discord.Status.offline and mem_after.status is discord.Status.online:
+            self.members_online.add(mem_after)
+            if self.timed_task.done():
+                self.timed_task = asyncio.create_task(self.timed_update())
+            print("added:", mem_after.name)
+        
+        elif mem_before.status is discord.Status.online and mem_after.status is discord.Status.offline:
+            self.members_online.remove(mem_before)
+            print("removed:", mem_before.name)
+            if len(self.members_online) <= 0:
+                self.timed_task.cancel()
+                try:
+                    await self.timed_task
+                except asyncio.CancelledError:
+                    print("task is now cancelled")
         
         # Track activity updates for members
         if not mem_before.activity and mem_after.activity: # activity starts/ongoing
-            if mem_after.name not in self.timers:
-                self.timers[mem_after.name] = {} # If the member is not already tracked, add them
-            if self.current_date not in self.timers[mem_after.name]:
-                self.timers[mem_after.name][self.current_date] = {} # If the current date is not present in the member's dictionary, add it
-            if mem_after.activity.name not in self.timers[mem_after.name][self.current_date]: # If the member's current activity/game/status is not present, add it with initial values
-                self.timers[mem_after.name][self.current_date][mem_after.activity.name] = {
+            after_string_id = str(mem_after.id)
+            if after_string_id not in self.timers:
+                self.timers[after_string_id] = {} # If the member is not already tracked, add them
+            if self.current_date not in self.timers[after_string_id]:
+                self.timers[after_string_id][self.current_date] = {} # If the current date is not present in the member's dictionary, add it
+            if mem_after.activity.name not in self.timers[after_string_id][self.current_date]: # If the member's current activity/game/status is not present, add it with initial values
+                self.timers[after_string_id][self.current_date][mem_after.activity.name] = {
```

Comment:
don't really know the best solution but looking at this i'm thinking about how after many days, we'll have extra "startTime" that we won't use anymore per person per game per day. is there maybe a way after each day passes that we just remove the startTime? a perfect solution would be to only have one startTime total but changing up the ordering if the dictionary feels even sloppier if we had dates after activities.